docs/proposals/20220411-cluster-api-state-metrics.md

```diff
@@ -197,31 +223,6 @@ These `customresource.RegistryFactory` implementations get used in a `main.go` w
   - imports and uses `k8s.io/kube-state-metrics/v2/pkg/options.NewOptions()`[[3]] to define the same cli flags and options as kube-state-metrics, except the enabled metrics.
   - imports and uses `k8s.io/kube-state-metrics/v2/pkg/app.RunKubeStateMetrics(...)`[[3]] to start the metrics server using the given options and the custom registry factory from `store.Factories()`.
 
-### Security Model
-
-- RBAC definitions should be generated via kubebuilder annotations.
-- RBAC definitions should only grant `get`, `list` and `watch` permissions to CRs relevant for the application.
-
-### Risks and Mitigations
-
-This initial implementation provides a baseline on which incremental changes can be introduced in the future. Instead of encompassing all possible use cases under a single proposal, this proposal mitigates the risk of waiting too long to consider all required use cases regarding this topic.
-
-## Alternatives
-
-### Directly use kube-state-metrics
-
-> [kube-state-metrics] is a simple service that listens to the Kubernetes API server and generates metrics about the state of the objects.
-
-On a first thought, using kube-state-metrics would be a great fit to retrieve the desired metrics. However, kube-state-metrics does not plan to implement metrics for CRs of CRDs:
-
-> There is no meaningful extension kube-state-metrics can do other than potentially providing a library to reuse the mechanisms built in this repository. [[4]]
-
-The linked issue also states that:
-
-> operators should expose metrics about the objects they expose themselves. [[4]]
-
-Because of that kube-state-metrics itself does not fit this use-case.
-
 ### Expose metrics by the controllers
 
 To solve this in the spirit of the kube-state-metrics maintainers it may make sense to implement the metrics directly within the CAPI controllers.
```

Comment:
This seems outdated given that with "the spirit of the kube-state-metrics maintainers" we meant they don't want to support CRDs.

(but we can just drop this part of the sentence)